src/hmem_cuda.c

```diff
@@ -442,18 +442,25 @@ static int cuda_hmem_dl_init(void)
 	cuda_attr.nvml_handle = dlopen("libnvidia-ml.so", RTLD_NOW);
 	if (!cuda_attr.nvml_handle) {
 		FI_WARN(&core_prov, FI_LOG_CORE,
-			"Failed to dlopen libnvidia-ml.so\n");
-		goto err_dlclose_cuda_driver;
+			"Failed to dlopen libnvidia-ml.so.  Trying libnvidia-ml.so.1\n");
+		cuda_attr.nvml_handle = dlopen("libnvidia-ml.so", RTLD_NOW);
+		if (!cuda_attr.nvml_handle) {
+			FI_WARN(&core_prov, FI_LOG_CORE,
+			"Failed to dlopen libnvidia-ml.so.1 also, bypassing nvml calls\n");
+		}
 	}
 
 	CUDA_DRIVER_FUNCS_DEF(CUDA_DRIVER_FUNCS_DLOPEN)
 	CUDA_RUNTIME_FUNCS_DEF(CUDA_RUNTIME_FUNCS_DLOPEN)
-	NVML_FUNCS_DEF(NVML_LIB_FUNCS_DLOPEN)
+	if (cuda_attr.nvml_handle) {
+		NVML_FUNCS_DEF(NVML_LIB_FUNCS_DLOPEN)
+	}
 
 	return FI_SUCCESS;
 
 err_dlclose_nvml_lib:
-	dlclose(cuda_attr.nvml_handle);
+	if (cuda_attr.nvml_handle)
+		dlclose(cuda_attr.nvml_handle);
```

Comment:
This entire label section does not appear to be accessible.

Reply:
The label is referenced within the CUDA_FUNCS_DLOPEN macro, within the *_FUNCS_DEF macros.

Reply:
got it -- thanks